x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -177,40 +198,45 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
     }
   }, [caseDetailsRefresh]);
 
-  if (!expandedEvent?.eventId) {
-    return null;
-  }
-
-  return isFlyoutView ? (
-    <>
-      <EuiFlyoutHeader hasBorder={isHostIsolationPanelOpen}>
-        {isHostIsolationPanelOpen ? (
-          backToAlertDetailsLink
-        ) : (
+  const renderFlyoutHeader = useMemo(() => {
+    let text;
+    switch (isActivePanel) {
+      case ACTIVE_PANEL.OSQUERY:
+        text = <h2>{RUN_OSQUERY}</h2>;
+        return backToAlertDetailsLink(text);
+      case ACTIVE_PANEL.HOST_ISOLATION:
+        text = <h2>{isolateAction === 'isolateHost' ? ISOLATE_HOST : UNISOLATE_HOST}</h2>;
+        return backToAlertDetailsLink(text);
+      default:
+        return (
           <ExpandableEventTitle
             isAlert={isAlert}
             loading={loading}
             ruleName={ruleName}
             timestamp={timestamp}
           />
-        )}
-      </EuiFlyoutHeader>
-      {isIsolateActionSuccessBannerVisible && (
-        <EndpointIsolateSuccess
-          hostName={hostName}
-          alertId={alertId}
-          isolateAction={isolateAction}
-        />
-      )}
-      <StyledEuiFlyoutBody>
-        {isHostIsolationPanelOpen ? (
+        );
+    }
+  }, [backToAlertDetailsLink, isActivePanel, isAlert, isolateAction, loading, ruleName, timestamp]);
+  const renderFlyoutBody = useMemo(() => {
```

Comment:
can we extract it to the standalone component?

Reply:
I would leave this as it is, there is too many dependencies within this component. And I do not want to pass tens of props down.